### PR TITLE
PreviewBadge with option to open in PreviewMode in BackOffice…

### PR DESCRIPTION
**Pull request by nielslyngsoe**
Option to open in PreviewMode in BackOffice plus new styling, now the badge will look like this, and be placed center in the bottom. Plus a little motion to grab attention without being annoying in general.

image

### CODE REVIEW

File: src/Umbraco.Web.UI/config/umbracoSettings.Release.config [hunk 1]
```diff
@@ -1,113 +1,245 @@
...
+                <span class="umbraco-preview-badge__header">Preview mode</span>
+                <a href="{0}/preview/?id={2}" class="umbraco-preview-badge__a open">
+                    …
+                </a>
+                <a href="{0}/preview/end?redir={1}" class="umbraco-preview-badge__a end">
+                    <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg"><title>Click to end</title><path fill="#fff" d="M5273.1 2400.1v-2c0-2.8-5-4-9.7-4s-9.7 1.3-9.7 4v2a7 7 0 002 4.9l5 4.9c.3.3.4.6.4 1v6.4c0 .4.2.7.6.8l2.9.9c.5.1 1-.2 1-.8v-7.2c0-.4.2-.7.4-1l5.1-5a7 7 0 002-4.9zm-9.7-.1c-4.8 0-7.4-1.3-7.5-1.8.1-.5 2.7-1.8 7.5-1.8s7.3 1.3 7.5 1.8c-.2.5-2.7 1.8-7.5 1.8z"/><path fill="#fff" d="M5268.4 2410.3c-.6 0-1 .4-1 1s.4 1 1 1h4.3c.6 0 1-.4 1-1s-.4-1-1-1h-4.3zM5272.7 2413.7h-4.3c-.6 0-1 .4-1 1s.4 1 1 1h4.3c.6 0 1-.4 1-1s-.4-1-1-1zM5272.7 2417h-4.3c-.6 0-1 .4-1 1s.4 1 1 1h4.3c.6 0 1-.4 1-1 0-.5-.4-1-1-1z"/><path fill="#fff" d="M78.2 13l-8.7 11.7a32.5 32.5 0 11-51.9 25.8c0-10.3 4.7-19.7 12.9-25.8L21.8 13a47 47 0 1056.4 0z"/><path fill="#fff" d="M42.7 2.5h14.6v49.4H42.7z"/></svg>
+                </a>
+            </div>
+            <style type="text/css">
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI/config/umbracoSettings.Release.config]: EEK @nielslyngsoe do we need to inline this CSS, could it not be in an external file. As this makes this XML config file not fun to work with

File: src/Umbraco.Web.UI/config/umbracoSettings.config [hunk 2]
```diff
@@ -1,112 +1,245 @@
...
+                <span class="umbraco-preview-badge__header">Preview mode</span>
+                <a href="{0}/preview/?id={2}" class="umbraco-preview-badge__a open">
+                    …
+                </a>
+                <a href="{0}/preview/end?redir={1}" class="umbraco-preview-badge__a end">
+                    <svg viewBox="0 0 100 100" xmlns="http://www.w3.org/2000/svg"><title>Click to end</title><path fill="#fff" d="M5273.1 2400.1v-2c0-2.8-5-4-9.7-4s-9.7 1.3-9.7 4v2a7 7 0 002 4.9l5 4.9c.3.3.4.6.4 1v6.4c0 .4.2.7.6.8l2.9.9c.5.1 1-.2 1-.8v-7.2c0-.4.2-.7.4-1l5.1-5a7 7 0 002-4.9zm-9.7-.1c-4.8 0-7.4-1.3-7.5-1.8.1-.5 2.7-1.8 7.5-1.8s7.3 1.3 7.5 1.8c-.2.5-2.7 1.8-7.5 1.8z"/><path fill="#fff" d="M5268.4 2410.3c-.6 0-1 .4-1 1s.4 1 1 1h4.3c.6 0 1-.4 1-1s-.4-1-1-1h-4.3zM5272.7 2413.7h-4.3c-.6 0-1 .4-1 1s.4 1 1 1h4.3c.6 0 1-.4 1-1s-.4-1-1-1zM5272.7 2417h-4.3c-.6 0-1 .4-1 1s.4 1 1 1h4.3c.6 0 1-.4 1-1 0-.5-.4-1-1-1z"/><path fill="#fff" d="M78.2 13l-8.7 11.7a32.5 32.5 0 11-51.9 25.8c0-10.3 4.7-19.7 12.9-25.8L21.8 13a47 47 0 1056.4 0z"/><path fill="#fff" d="M42.7 2.5h14.6v49.4H42.7z"/></svg>
+                </a>
+            </div>
+            <style type="text/css">
```
Comment (warrenbuckley) [inline src/Umbraco.Web.UI/config/umbracoSettings.config]: Same again re the inline CSS <style>

File: src/Umbraco.Web.UI/config/umbracoSettings.Release.config (hunk 1 above)
Comment (nielslyngsoe) [reply src/Umbraco.Web.UI/config/umbracoSettings.Release.config]: I agree, but the scope of this task was just to change the badge, not the architecture on how its made or configured — I also think that would be a breaking change.

**Comment by warrenbuckley** [comment]
Agreed to disagree on this in regards if the SVG and CSS should be made external, but for ease of use and to avoid loading more resources it was done inline.

Off to test this out...

**Comment by warrenbuckley** [comment]
Works nicely by being in the middle of the page at the bottom with the nice animation to grab attention.

**Pull request merged**